samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -165,18 +220,47 @@ class DailyCallManager {
   }
 
   /**
-   * Removes the track's element (audio or video) from the DOM when it stops
+   * Removes a media element (audio or video) from the DOM for a specific participant
    * and ensures media resources are released by setting srcObject to null.
+   * @param {string} participantId - The ID of the participant.
+   * @param {string} trackKind - The kind of track ('video' or 'audio').
+   */
+  removeMediaElement(participantId, trackKind) {
+    // Construct the selector based on the track kind
+    let elementSelector;
+    if (trackKind === 'video') {
+      elementSelector = `#video-container-${participantId} video`;
+    } else if (trackKind === 'audio') {
+      elementSelector = 'audio';
+    } else {
+      console.error('Unsupported track kind:', trackKind);
+      return;
+    }
+
+    const element = document.querySelector(elementSelector);
+    if (element) {
+      element.srcObject = null; // Release media resources
+      element.remove(); // Remove the element from the DOM
+    }
+  }
+
+  /**
+   * Adjusted destroyTrack method that calls removeMediaElement for each track.
    * @param {Object} event - The track event object.
    */
   destroyTrack(event) {
-    let tracks = document.querySelectorAll('video, audio');
-    tracks.forEach((el) => {
-      if (el.srcObject && el.srcObject.getTracks().includes(event.track)) {
-        el.srcObject = null; // Release media resources
-        el.remove(); // Remove the element from the DOM
-      }
-    });
+    const participantId = event.participant && event.participant.session_id;
+    // Ensure participantId exists for video tracks, or use a placeholder for audio
+    const effectiveParticipantId =
+      event.track.kind === 'video' ? participantId : 'localAudio';
+    if (effectiveParticipantId) {
+      this.removeMediaElement(effectiveParticipantId, event.track.kind);
+    } else {
+      console.error(
```

Comment:
Sort of related to my question about participants leaving, I am consistently getting this error when a participant leaves. This makes me think that the participant leave event is firing before the track stopped event, which I suspect also results in the associated WebMediaPlayer only being paused, not destroyed: 

<img width="827" alt="image" src="https://github.com/daily-demos/daily-samples-js/assets/65890040/fd36bc13-12c0-4884-8ac6-b25a02f61d61">

Reply:
You're right. This isn't working upon leaving. The root cause appears to be a bug in daily-js, which I just discovered thanks to your nudging: https://linear.app/dailyco/issue/CSDK-2110/when-a-track-stopped-event-is-emitted-due-to-a-participant-leaving-a.

I'm inclined to leave this in here and just fixing this issue in daily-js. With the participant information missing from track-stopped, the simplicity of `track-stopped` is broken. This forces someone to rely on `participant-updated` to get the full event. Anyway, thanks for finding this issue! 🙌

Reply:
Makes sense! Approving